src/jit/lsraxarch.cpp

```diff
@@ -2314,6 +2314,19 @@ void LinearScan::BuildHWIntrinsic(GenTreeHWIntrinsic* intrinsicTree)
         }
     }
 
+    // Check for "srcCount >= 2" to match against 3+ operand nodes where one is constant
+    if ((op2 == nullptr) && (info->srcCount >= 2) && intrinsicTree->isRMWHWIntrinsic(compiler))
+    {
+        // TODO-XArch-CQ: This is currently done in order to handle intrinsics which have more than
+        // two arguments but which still have RMW semantics (such as NI_SSE41_Insert). We should make
+        // this handling more general and move it back out to LinearScan::BuildNode.
```

Comment:
Actually, the RMW handling will be changing somewhat, so I wouldn't hurry to make this change.